### PR TITLE
KFSTP-1802

### PR DESCRIPTION
**Pull request by smith750**
Rename fields to match what column labels are in java

### CODE REVIEW

**Comment by kkronenb** [comment]
This is going to be confusing now that the names are not obfuscated....

I've taken a quick look and it makes sense to me, but I'll let Bryan do the merge.

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 1]
```diff
@@ -101,57 +101,57 @@ public void setDocumentNumber(String documentNumber) {
...
     /**
-     * Sets the budget attribute value.
+     * Sets the totalBudget attribute value.
      *
-     * @param budget The budget to set.
+     * @param totalBudget The totalBudget to set.
      */
-    public void setBudget(KualiDecimal budget) {
-        this.budget = budget;
+    public void setTotalBudget(KualiDecimal budget) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: this is nitpicky, but do you want to change the parm name too?

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 2]
```diff
@@ -101,57 +101,57 @@ public void setDocumentNumber(String documentNumber) {
...
     /**
-     * Sets the expenditures attribute value.
+     * Sets the invoiceAmount attribute value.
      *
-     * @param expenditures The expenditures to set.
+     * @param invoiceAmount The invoiceAmount to set.
      */
-    public void setExpenditures(KualiDecimal expenditures) {
-        this.expenditures = expenditures;
+    public void setInvoiceAmount(KualiDecimal expenditures) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: parm name here too?

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 3]
```diff
@@ -101,57 +101,57 @@ public void setDocumentNumber(String documentNumber) {
...
     /**
-     * Sets the cumulative attribute value.
+     * Sets the cumulativeExpenditures attribute value.
      *
-     * @param cumulative The cumulative to set.
+     * @param cumulativeExpenditures The cumulativeExpenditures to set.
      */
-    public void setCumulative(KualiDecimal cumulative) {
-        this.cumulative = cumulative;
+    public void setCumulativeExpenditures(KualiDecimal cumulative) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: and again?

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 4]
```diff
@@ -159,29 +159,29 @@ public void setCumulative(KualiDecimal cumulative) {
      *
      * @return Returns the balance.
      */
-    public KualiDecimal getBalance() {
+    public KualiDecimal getBudgetRemaining() {
         // Balance = Budget-Cumulative
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: comment is a bit out of date; I'd probably remove altogether and let the code speak for itself

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 5]
```diff
@@ -159,29 +159,29 @@ public void setCumulative(KualiDecimal cumulative) {
      *
      * @return Returns the balance.
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: update comment?

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 6]
```diff
@@ -159,29 +159,29 @@ public void setCumulative(KualiDecimal cumulative) {
...
     /**
-     * Sets the billed attribute value.
+     * Sets the totalPreviouslyBilled attribute value.
      *
-     * @param billed The billed to set.
+     * @param totalPreviouslyBilled The totalPreviouslyBilled to set.
      */
-    public void setBilled(KualiDecimal billed) {
-        this.billed = billed;
+    public void setTotalPreviouslyBilled(KualiDecimal billed) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: rename parm again

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 7]
```diff
@@ -236,9 +236,9 @@ public void setCostCategory(CostCategory costCategory) {
      *
      * @return Returns the adjustedCumExpenditures.
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: fix comment

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 8]
```diff
@@ -247,9 +247,9 @@ public KualiDecimal getAdjustedCumExpenditures() {
      *
      * @return Returns the adjustedBalance.
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: fix comment

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 9]
```diff
@@ -264,21 +264,21 @@ protected LinkedHashMap toStringMapper_RICE20_REFACTORME() {
...
         }
-        if (ObjectUtils.isNotNull(this.cumulative)) {
-            m.put("cumulative", this.cumulative.toString());
+        if (ObjectUtils.isNotNull(this.cumulativeExpenditures)) {
+            m.put("cumulativeExpenditures", this.cumulativeExpenditures.toString());
         }
-        m.put("balance", getBalance().toString());
-        if (ObjectUtils.isNotNull(this.billed)) {
-            m.put("billed", this.billed.toString());
+        m.put("balance", getBudgetRemaining().toString());
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: "balance" -> "budgetRemaining"?

File: work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java [hunk 10]
```diff
@@ -264,21 +264,21 @@ protected LinkedHashMap toStringMapper_RICE20_REFACTORME() {
...
+        if (ObjectUtils.isNotNull(this.totalPreviouslyBilled)) {
+            m.put("totalPreviouslyBilled", this.totalPreviouslyBilled.toString());
         }
-        m.put("adjustedCumExpenditures", getAdjustedCumExpenditures().toString());
-        m.put("adjustedBalance", getAdjustedBalance().toString());
+        m.put("adjustedCumExpenditures", getTotalAmountBilledToDate().toString());
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/ContractsGrantsInvoiceDetail.java]: rename keys on this line and the next?

File: work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java [hunk 11]
```diff
@@ -140,59 +140,59 @@ public void setContractControlAccountNumber(String contractControlAccountNumber)
...
-     * Sets the budgetAmount attribute value.
+     * Sets the totalBudget attribute value.
      *
-     * @param budgetAmount The budgetAmount to set.
+     * @param totalBudget The totalBudget to set.
      */
 
-    public void setBudgetAmount(KualiDecimal budgetAmount) {
-        this.budgetAmount = budgetAmount;
+    public void setTotalBudget(KualiDecimal budgetAmount) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java]: rename parm

File: work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java [hunk 12]
```diff
@@ -140,59 +140,59 @@ public void setContractControlAccountNumber(String contractControlAccountNumber)
...
     /**
-     * Sets the expenditureAmount attribute value.
+     * Sets the invoiceAmount attribute value.
      *
-     * @param expenditureAmount The expenditureAmount to set.
+     * @param invoiceAmount The invoiceAmount to set.
      */
-    public void setExpenditureAmount(KualiDecimal expenditureAmount) {
-        this.expenditureAmount = expenditureAmount;
+    public void setInvoiceAmount(KualiDecimal expenditureAmount) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java]: rename parm

File: work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java [hunk 13]
```diff
@@ -140,59 +140,59 @@ public void setContractControlAccountNumber(String contractControlAccountNumber)
...
-     * Sets the cumulativeAmount attribute value.
+     * Sets the cumulativeExpenditures attribute value.
      *
-     * @param cumulativeAmount The cumulativeAmount to set.
+     * @param cumulativeExpenditures The cumulativeExpenditures to set.
      */
 
-    public void setCumulativeAmount(KualiDecimal cumulativeAmount) {
-        this.cumulativeAmount = cumulativeAmount;
+    public void setCumulativeExpenditures(KualiDecimal cumulativeAmount) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java]: rename parm

File: work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java [hunk 14]
```diff
@@ -224,12 +224,12 @@ public void setInvoiceDocument(ContractsGrantsInvoiceDocument invoiceDocument) {
...
 
-    public KualiDecimal getBilledAmount() {
-        return billedAmount;
+    public KualiDecimal getTotalAmountBilledToDate() {
+        return totalAmountBilledToDate;
     }
 
-    public void setBilledAmount(KualiDecimal billedAmount) {
-        this.billedAmount = billedAmount;
+    public void setTotalAmountBilledToDate(KualiDecimal billedAmount) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/InvoiceAccountDetail.java]: rename parm

File: work/src/org/kuali/kfs/module/ar/businessobject/InvoiceGeneralDetail.java [hunk 15]
```diff
@@ -215,22 +215,22 @@ public void setAwardTotal(KualiDecimal awardTotal) {
...
     /**
-     * Sets the newTotalBilled attribute value.
+     * Sets the totalAmountBilledToDate attribute value.
      *
-     * @param newTotalBilled The newTotalBilled to set.
+     * @param totalAmountBilledToDate The totalAmountBilledToDate to set.
      */
-    public void setNewTotalBilled(KualiDecimal newTotalBilled) {
-        this.newTotalBilled = newTotalBilled;
+    public void setTotalAmountBilledToDate(KualiDecimal newTotalBilled) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/InvoiceGeneralDetail.java]: rename parm

File: work/src/org/kuali/kfs/module/ar/businessobject/InvoiceGeneralDetail.java [hunk 16]
```diff
@@ -293,21 +293,21 @@ protected LinkedHashMap toStringMapper_RICE20_REFACTORME() {
...
     /**
-     * Sets the billedToDateAmount attribute value.
+     * Sets the totalPreviouslyBilled attribute value.
      *
-     * @param billedToDateAmount The billedToDateAmount to set.
+     * @param totalPreviouslyBilled The totalPreviouslyBilled to set.
      */
-    public void setBilledToDateAmount(KualiDecimal billedToDateAmount) {
-        this.billedToDateAmount = billedToDateAmount;
+    public void setTotalPreviouslyBilled(KualiDecimal billedToDateAmount) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/businessobject/InvoiceGeneralDetail.java]: rename parm

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java [hunk 17]
```diff
@@ -495,41 +495,41 @@ public String getCustomerNumber() {
...
+            total = total.add(accountDetail.getBudgetRemaining());
         }
         return total;
     }
 
-    public KualiDecimal getBilledAmountTotal() {
+    public KualiDecimal getTotalAmountBilledToDate() {
         KualiDecimal total = KualiDecimal.ZERO;
         for (InvoiceAccountDetail accountDetail : accountDetails) {
             total = total.add(accountDetail.getAdjustedCumExpenditures());
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: should the accountDetail.getAdjustedCumExpenditures() method be renamed to getTotalAmountBilledToDate()?

File: work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java [hunk 18]
```diff
@@ -1205,13 +1205,13 @@ public void generateInvoicesForInvoiceAddresses(ContractsGrantsInvoiceDocument d
...
-                parameterMap.put(ArPropertyConstants.INVOICE_DETAIL+"[" + i + "]."+ArPropertyConstants.CUMULATIVE, document.getDirectCostInvoiceDetails().get(i).getCumulative());
-                parameterMap.put(ArPropertyConstants.INVOICE_DETAIL+"[" + i + "]."+ArPropertyConstants.BALANCE, document.getDirectCostInvoiceDetails().get(i).getBalance());
-                parameterMap.put(ArPropertyConstants.INVOICE_DETAIL+"[" + i + "]."+ArPropertyConstants.BILLED, document.getDirectCostInvoiceDetails().get(i).getBilled());
-                parameterMap.put(ArPropertyConstants.INVOICE_DETAIL+"[" + i + "]."+ArPropertyConstants.ADJUSTED_CUMULATIVE_EXPENDITURES, document.getDirectCostInvoiceDetails().get(i).getAdjustedCumExpenditures());
-                parameterMap.put(ArPropertyConstants.INVOICE_DETAIL+"[" + i + "]."+ArPropertyConstants.ADJUSTED_BALANCE, firstInvoiceDetail.getAdjustedBalance());
+                parameterMap.put(ArPropertyConstants.INVOICE_DETAIL+"[" + i + "]."+KFSPropertyConstants.BUDGET, document.getDirectCostInvoiceDetails().get(i).getTotalBudget());
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: do these constants need to be renamed?

File: work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java [hunk 19]
```diff
@@ -1150,14 +1150,14 @@ protected KualiDecimal calculateTotalExpenditureAmount(ContractsGrantsInvoiceDoc
...
+                            invAcctD.setInvoiceAmount(currentExpenditureAmount);
                         }
                     }
                 }
             }
             totalExpendituredAmount = totalExpendituredAmount.add(currentExpenditureAmount);
         }
-        totalExpendituredAmount = totalExpendituredAmount.add(document.getInvoiceGeneralDetail().getBilledToDateAmount());
+        totalExpendituredAmount = totalExpendituredAmount.add(document.getInvoiceGeneralDetail().getTotalPreviouslyBilled());
         return totalExpendituredAmount;
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java]: should totalExpenditureAmount be totalInvoiceAmount?

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java (hunk 17 above)
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: We can't; InvoiceAccountDetail already has a totalAmountBilledToDate property.  Now, why _those_ aren't being summed here...not at all sure.

File: work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java (hunk 19 above)
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java]: this one gets a little wierd...let's discuss...

File: work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java (hunk 18 above)
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: Are we going to need to update templates as well?

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java (hunk 17 above)
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: Discussed; let's rename totalAmountBilledToDate to totalPreviouslyBilled and rename adjustedCumulativeExpenditures to totalAmountBilledToDate.

File: work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java (hunk 18 above)
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: I'll ask kfs-chatter if we're going to deliver templates out the box.

**Pull request merged**